R/bootstrap_traits.R

```diff
@@ -22,22 +23,31 @@
 #' boot_traits <- trait_np_bootstrap(imputed_traits)
 #' @export
 
-trait_np_bootstrap <- function(imputed_traits, nrep = 100, sample_size = 200) {
+trait_np_bootstrap <- function(imputed_traits, nrep = 100, sample_size = 200, raw = FALSE) {
+  if (raw) {nrep <- 1}
   #  stopifnot(class(traits_com) == "imputed_traits")
   attrib <- attr(imputed_traits, "attrib")
   value_col <- attrib$value_col
   bootstrap_moments <- map_dfr(
     1:nrep,
     ~ {
-      slice_sample(imputed_traits, n = sample_size,
-                   replace = TRUE, weight_by = weight) %>%
+      raw_dist <- slice_sample(imputed_traits, n = sample_size,
+                               replace = TRUE, weight_by = weight)
+      if (raw){
+        return(raw_dist)
+      }
+      
+      else {
```

Comment:
To my surprise, this works, but is unusual and will not run interactively. Better to use 
`
if (raw) {
 ...
} else {
  ...
}
`

Also in the other function